alf/experience_replayers/replay_buffer.py

```diff
@@ -254,7 +360,164 @@ def _prioritized_sample(self, batch_size, batch_length):
         r = (r + torch.arange(batch_size, dtype=torch.float32)) / batch_size
         r = r * self._sum_tree.summary()
         indices = self._sum_tree.find_sum_bound(r)
-        return self._index_to_env_id_pos(indices)
+        env_ids, idx = self._index_to_env_id_idx(indices)
+        return env_ids, idx
+
+    def _pad(self, x, env_ids):
+        """Make ``x`` (any index) absolute positions in the RingBuffer.
+
+        This is the reverse of ``circular()``, and is useful when trying to
+        compute the distance from one index to the other.  This is done by
+        adding multiples of _max_length back to the index.
+
+        NOTE, this operation depends on the _current_pos of the RingBuffer,
+        and can generate different results if new data are added to the buffer.
+        """
```

Comment:
position can be calculated more efficiently:
position = idx + n L
current_pos - L <= idx + n L < current_pos
current_pos - idx - 1 - L < n L <= current_pos - idx - 1
n = (current_pos - idx - 1) / L
position =  ((current_pos - idx - 1) / L) * L + idx

Reply:
simplify as above